### PR TITLE
[bug fix] assigning the reurn value of new by referece is now deprecated.

### PR DESCRIPTION
**Pull request by mahone3297**

### CODE REVIEW

File: src/Visitor/Usage/FindLanguageDeprecations.php [hunk 1]
```diff
@@ -33,9 +33,11 @@ public function setPhpFileInfo(PhpFileInfo $phpFileInfo)
     public function enterNode(Node $node)
     {
         if ($node instanceof Node\Expr\AssignRef) {
-            $this->phpFileInfo->addDeprecatedLanguageUsage(
-                new DeprecatedLanguageUsage('assign by reference(&=)', 'Since PHP 5.3 use normal assignment instead.', $node->getLine())
-            );
+            if ($node->expr instanceof Node\Expr\New_) {
```
Comment (xabbuh) [inline src/Visitor/Usage/FindLanguageDeprecations.php]: I would not nest the `if` statements here, but combine them instead like this:

```php
if ($node instanceof Node\Expr\AssignRef && $node->expr instanceof Node\Expr\New_) {
    // ...
}
```

Comment (mahone3297) [inline src/Visitor/Usage/FindLanguageDeprecations.php]: First, I change the code with "combine" style what you said.
And then, I think if use "combine" style, some cases need to change this logic, more lines need to update compare to "nest" style. So I change it to "combine" style.

Finally, I will change the doc follow you said "combine" style.

**Comment by mahone3297** [comment]
@MarvinKlemp Hi, Can you review and merge this PR if it is ok?
looking forward to your reply.

**Comment by slde-robin** [comment]
@mahone3297 looks good to me :)

**Pull request merged**